dj/models/node.py

```diff
@@ -466,24 +466,27 @@ def format_metric_alias(query: str, name: str) -> str:
 
         tree = parse(query)
         projection_0 = tree.select.projection[0]
-        # the expression has to be something we can get a name out of
-        if not hasattr(projection_0, "alias_or_name"):
-            raise DJInvalidInputException(
-                f"Metric expression of type {type(projection_0).__name__} "
-                f"must be aliased as the node name: `{amenable_name(name)}`",
-            )
+
         # if the name is not what we expect, check if it is an alias
         # if it is an alias, we will raise because the user will have
         # deliberately named this expression
         # otherwise, we will just add the alias we want e.g. the node name
-        expr_name: ast.Name = projection_0.alias_or_name  # type: ignore
-        if expr_name.name != amenable_name(name):
-            if expr_name.parent_key == "alias":
-                raise DJInvalidInputException(
-                    "Invalid Metric. The expression in the projection "
-                    "cannot have alias different from the node name. Got "
-                    f"`{expr_name}` but expected `{amenable_name(name)}`",
-                )
+        expr_name: Optional[ast.Name] = (
```

Comment:
@CircArgs I changed the logic for this so that any metric expression, regardless of whether it's a binaryop or a single column, as long as it doesn't have an alias, will get an alias set with this method. Otherwise I think users will be confused as to why their complex metric expression doesn't get auto-assigned an alias.